src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -933,61 +936,97 @@ void SQLStorage::clearInstallationResult() {
   }
 }
 
+boost::optional<std::pair<int64_t, size_t>> SQLStorage::isTargetFileExists(const Uptane::Target& target) const {
+  std::lock_guard<std::mutex> lock(sql_mutex);
+  SQLite3Guard db = dbConnection();
+
+  auto statement = db.prepareStatement<std::string>(
+      "SELECT rowid, real_size, sha256, sha512 FROM target_images WHERE filename = ?;", target.filename());
+
+  int statement_state;
+  while ((statement_state = statement.step()) == SQLITE_ROW) {
+    auto sha256 = statement.get_result_col_str(2);
+    auto sha512 = statement.get_result_col_str(3);
+    if ((*sha256).empty() && (*sha512).empty()) {
+      LOG_WARNING << "Image without checksum: " << target.filename();
+      continue;
+    }
```

Comment:
This will probably be a problem for anyone who upgrades to this version of aktualizr with a database with targets stored before we added these hashes. We either will have to manually add those hashes while migrating or accept targets without any hashes. Or perhaps some way to indicate a legacy target could work.

Reply:
After our discussion, I agree that migration compatibility is actually not a problem. The worst that would happen is the user would have to re-download a package, but that's actually quite unlikely to be possible or necessary. Perhaps leaving a comment somewhere to explain that is a good idea, though.